network/ping_test.go

```diff
@@ -60,7 +60,7 @@ func TestPing(t *testing.T) {
 		if lastPingRoundTripTime > 0 {
 			postPing := time.Now()
 			testTime := postPing.Sub(prePing)
-			if (lastPingRoundTripTime < testTime) && (lastPingRoundTripTime > (testTime - (2 * waitStep))) {
+			if (lastPingRoundTripTime < testTime) && (lastPingRoundTripTime > (testTime - ((time.Duration(i) * time.Second) * waitStep))) {
```

Comment:
Could you please explain why this was a problem to start with, and why running with mule requires this ?

Reply:
This test failed a lot in lower compute environments because the testTime would be very long. I thought having a gradually increasing wait time would be better, but it's possible I'm missing exactly what this is trying to exercise

Reply:
Could you please have @algobolson review this change ? He wrote this unit test, and could suggest what the best course of action would be.

Reply:
This is perhaps a slightly foolish timing assertion. The first half `(lastPingRoundTripTime < testTime)` says that the measured ping round trip time is less than the time observed at the test from before the ping is sent until after the reply is received. If this fails then ping time is being measured wrong or the test system clock is insane.
The second half `(lastPingRoundTripTime > (testTime - (2 * waitStep)))` says that the polling for ping reply didn't wait more than 2x the polling period to get the reply. _maybe this doesn't matter and we should just delete this half of the expression_ ?